### PR TITLE
Don't ignore errors from failing ImageWriter construction

### PR DESCRIPTION
**Pull request by JonathanMeans**
Fixes https://github.com/mdaus/nitro/issues/188

### CODE REVIEW

File: modules/c++/nitf/source/ImageWriter.cpp [hunk 1]
```diff
@@ -56,14 +53,19 @@ void ImageWriter::setWriteCaching(int enable)
...
     {
         throw nitf::NITFException(&error);
     }
 }
 
 void ImageWriter::setPadPixel(nitf::Uint8* value, nitf::Uint32 length)
 {
-    if (!nitf_ImageWriter_setPadPixel(getNativeOrThrow(), value, length, &error))
+    if (!nitf_ImageWriter_setPadPixel(
+                getNativeOrThrow(), value, length, &error))
```
Comment (bfisc) [inline modules/c++/nitf/source/ImageWriter.cpp]: why the different style formatting between these two? (line 56 vs 55)

File: modules/c++/nitf/source/ImageWriter.cpp [hunk 2]
```diff
@@ -26,27 +26,24 @@ using namespace nitf;
 
 ImageWriter::ImageWriter(nitf::ImageSubheader& subheader)
 {
-    setNative(nitf_ImageWriter_construct(subheader.getNative(), NULL, &error));
+    auto* writer =
+            nitf_ImageWriter_construct(subheader.getNative(), NULL, &error);
+    setNativeOrThrow(writer, &error);
```
Comment (AndrewOfC) [inline modules/c++/nitf/source/ImageWriter.cpp]: It's not that we're getting a null native handle but it's failing in the setup.   The error member needs to be checked when nitf_ImageWriter_construct returns

```suggestion
    setNativeOrThrow(writer, &error);
    if( error.level != 0 )
       throw nitf::NITFException(error);
```

File: modules/c++/nitf/source/ImageWriter.cpp (hunk 1 above)
Comment (JonathanMeans) [reply modules/c++/nitf/source/ImageWriter.cpp]: I added an .clang-format auto-formatting script a while back. I'm gradually applying it to files as I touch them.

File: modules/c++/nitf/source/ImageWriter.cpp (hunk 2 above)
Comment (JonathanMeans) [reply modules/c++/nitf/source/ImageWriter.cpp]: It should be the case that `nitf_ImageWriter_construct` returns NULL if the error is set. I've updated that function to that effect.

**Pull request merged**